src/transac/components/top-bar/top-bar.store.coffee

```diff
@@ -0,0 +1,99 @@
+###
+#   @desc Store for TopBar Filters & menus state management
+###
+angular.module('transac.top-bar').service('TransacTopBarStore', ($q, MENUS, FILTERS_MENU)->
+
+  _self = @
+
+  state =
+    menus: angular.copy(MENUS)
+    selectedMenu: _.find(MENUS, 'active')
+    filtersMenu: angular.copy(FILTERS_MENU)
+    filters: {}
+    lastSearchQuery: ''
+
+  callbacks = {}
+  callbacks.dispatched = $q.defer()
+
+  ###
+  #   @desc Subscribe to notifications on state change via the dispatch method
+  #   @returns {Promise} A promise to the latest state
+  ###
+  @subscribe = ->
+    callbacks.dispatched.promise
+
+  ###
+  #   @desc Dispatch actions mutating the filters & menus state managed by this Store.
+  ###
+  @dispatch = (action, payload=null)->
+    switch action
+      when 'selectMenu'
+        _.each(state.menus, (menu) ->
+          menu.active = false
+          return
+        )
+        payload.active = true
+        state.selectedMenu = payload
+      when 'updateMenuCount'
+        menu = _.find(state.menus, 'type', payload.menuType)
+        menu.itemsCount = payload.itemsCount
+      when 'updateSearchFilter'
+        state.filters.$filter = buildFilterQuery(payload)
+      when 'selectFilter'
+        # Only one $orderby filter can be selected, and atleast one is always selected
+        if payload.type == '$orderby'
+          filters = _.filter(state.filtersMenu, 'type', payload.type)
+          _.each(filters, (f)-> f.selected = false & undefined)
```

Comment:
Yes, this looks like a mistake. The `_.each` method breaks when false is returned, which annoying with coffeescript one liners.